File: tests/benchmarks/resolution/fixtures/typescript/repository.ts
import type { Repository, User } from './types';

export class UserRepository implements Repository<User> {
  private store: Map<string, User> = new Map();

  findById(id: string): User | null {
    return this.store.get(id) ?? null;
  }

  save(user: User): void {
    this.store.set(user.id, user);
  }

  delete(id: string): boolean {
    return this.store.delete(id);
  }
}

export function createRepository(): Repository<User> {
  return new UserRepository();
}

File: tests/benchmarks/resolution/fixtures/objc/Repository.m
#import "Repository.h"

@implementation UserRepository {
    NSMutableDictionary *_store;
}

- (instancetype)init {
    self = [super init];
    if (self) {
        _store = [NSMutableDictionary dictionary];
    }
    return self;
}

- (void)saveWithId:(NSString *)userId name:(NSString *)name {
    _store[userId] = name;
}

- (NSString *)findById:(NSString *)userId {
    return _store[userId];
}

- (BOOL)deleteWithId:(NSString *)userId {
    if (_store[userId]) {
        [_store removeObjectForKey:userId];
        return YES;
    }
    return NO;
}

@end
